faunadb-java-dsl/src/main/java/com/faunadb/client/types/Value.java

```diff
@@ -75,14 +77,61 @@ private Value() {
   }
 
   /**
-   * Attempts to coerce this value using the {@link Codec} passed
+   * Attempts to coerce this value using the {@link Codec} passed.
    *
    * @param codec codec function to attempt coercion
    * @return the {@link Result} of the coercion
    * @see Codec
    */
   public final <T> Result<T> to(Codec<T> codec) {
-    return codec.apply(this);
+    return codec.decode(this);
+  }
+
+  /**
+   * Attempts to decode this value using the {@link Decoder} into the specified type.
+   *
+   * @param clazz a class type to convert
+   * @return the {@link Result} of the coercion
+   * @see Decoder
+   */
+  public final <T> Result<T> to(Class<T> clazz) {
+    return Decoder.decode(this, Types.of(clazz));
+  }
+
+  /**
+   * Attempts to encode an object using an {@link Encoder} into a {@link Value} type.
+   *
+   * @param obj the object instance to convert
+   * @return the {@link Result} of the conversion
+   */
+  public static <T> Result<Value> from(T obj) {
+    return Encoder.encode(obj);
+  }
+
+  /**
+   * Attempts to coerce this value to a {@link Map}.
+   * This method is only util if this object is an instance of {@link ObjectV}.
```

Comment:
"This method is only util if this object is an instance of {@link ObjectV}." can be rewritten as:

"This object must be an instance of {@link ObjectV}."